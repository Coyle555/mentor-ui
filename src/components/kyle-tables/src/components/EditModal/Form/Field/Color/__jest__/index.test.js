jest.mock('react-color', () => {
	return { GithubPicker: props => <div>{JSON.stringify(props)}</div> };
});

import React from 'react';
import { ColorField } from '../index';
import renderer from 'react-test-renderer';
import { cleanup, fireEvent, render } from '@testing-library/react';

afterEach(cleanup);

test('Color picker with a color passed in', () => {
	const tree = renderer.create(<ColorField value="#000000" />).toJSON();

	expect(tree).toMatchSnapshot();
});

test('Color picker with a new color passed in', () => {
	const tree = renderer.create(<ColorField value="#000000" />);
	tree.update(<ColorField value="#ffffff" />);

	expect(tree).toMatchSnapshot();
});

test('On color change callback', () => {
	const onColorChange = jest.fn();
	const instance = renderer.create(
		<ColorField
			fieldId="foo"
			onColorChange={onColorChange}
			rowId="baz"
			value="#000000"
		/>
	).getInstance();

	instance.onChangeComplete({ hex: '#00ffff' });
	expect(onColorChange).toHaveBeenCalledWith('baz', 'foo', '#00ffff');
});
