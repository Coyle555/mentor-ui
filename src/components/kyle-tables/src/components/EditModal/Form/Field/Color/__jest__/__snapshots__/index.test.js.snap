// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color picker with a color passed in 1`] = `
<div
  className="color-input-container"
>
  <div
    className="color-circle"
    style={
      Object {
        "backgroundColor": "#000000",
      }
    }
  />
  <div>
    {"color":"#000000","colors":["#000000","#FFFFFF","#B80000","#DB3E00","#FCCB00","#008B02","#006B76","#1273DE","#004DCF","#5300EB","#EB9694","#FAD0C3","#FEF3BD","#C1E1C5","#BEDADC","#C4DEF6","#BED3F3","#D4C4FB","#3D4B54","#001528","#008860","#004437","#EE104E","#A5F944","#003366","#F6A230","#4F054F","#008000","#0000FF","#FFFF00"],"width":"75%","triangle":"hide"}
  </div>
</div>
`;

exports[`Color picker with a new color passed in 1`] = `
<div
  className="color-input-container"
>
  <div
    className="color-circle"
    style={
      Object {
        "backgroundColor": "#ffffff",
      }
    }
  />
  <div>
    {"color":"#ffffff","colors":["#000000","#FFFFFF","#B80000","#DB3E00","#FCCB00","#008B02","#006B76","#1273DE","#004DCF","#5300EB","#EB9694","#FAD0C3","#FEF3BD","#C1E1C5","#BEDADC","#C4DEF6","#BED3F3","#D4C4FB","#3D4B54","#001528","#008860","#004437","#EE104E","#A5F944","#003366","#F6A230","#4F054F","#008000","#0000FF","#FFFF00"],"width":"75%","triangle":"hide"}
  </div>
</div>
`;
